src/Common/tendency.jl

```diff
@@ -155,6 +172,48 @@ end
     # TODO...
 end
 
+@inline function cloudy_precompute_aux_precip_helper(cloudy_params, moments, old_pdists)
+
+    FT = eltype(old_pdists[1])
+    # update the ParticleDistributions
+    mom_normed = tuple(moments ./ cloudy_params.mom_norms...)
+    pdists = ntuple(length(old_pdists)) do i
+        ind_rng = CL.get_dist_moments_ind_range(cloudy_params.NProgMoms, i)
+        args = (old_pdists[i], mom_normed[ind_rng])
+        if old_pdists[i] isa CL.ParticleDistributions.GammaPrimitiveParticleDistribution
+            CL.ParticleDistributions.update_dist_from_moments(args..., param_range = (; :k => (0.1, 10.0)))
+        else
+            CL.ParticleDistributions.update_dist_from_moments(args...)
+        end
+    end
+
+    # compute terminal velocity
+    sed_flux = -1 .* CL.Sedimentation.get_sedimentation_flux(pdists, cloudy_params.vel)
+    weighted_vt = ntuple(length(moments)) do i
+        ifelse(mom_normed[i] > FT(0), sed_flux[i] / mom_normed[i], FT(0))
+    end
+
+    return (; pdists, weighted_vt)
+end
+@inline function precompute_aux_precip!(ps::CloudyPrecip, Y, aux)
+
+    (; cloudy_params) = aux
+    (; ρ) = aux.thermo_variables
+    (; q_rai, N_rai, N_liq, pdists, moments) = aux.microph_variables
+    (; weighted_vt) = aux.velocities
+
+    rain_number_ind = Int(CL.get_dist_moments_ind_range(cloudy_params.NProgMoms, 2)[1])
+    rain_mass_ind = Int(CL.get_dist_moments_ind_range(cloudy_params.NProgMoms, 2)[2])
+    @. N_liq = Y.moments.:1
+    @. N_rai = Y.moments.:($$rain_number_ind)
+    @. q_rai = q_(Y.moments.:($$rain_mass_ind), ρ)
+    @. moments = Y.moments
+
+    tmp = @. cloudy_precompute_aux_precip_helper(cloudy_params, moments, pdists)
```

Comment:
I'm not very good at this, but I think that the way the Cloudy tendencies are added, both here and in 1D model, is allocating. Maybe instead we could create additional variables that we need for Cloudy and put them in `scratch` for temporary things that need to be stored during computations (like supersaturation, maybe `pdists`?) and in `aux` for the ones that need to be precomputed for tendencies. 

I was trying to write all the other tendencies like

```
@inline function precompute_aux_important_sources!
    source_term = aux.scratch.tmp
    (; stored_variable) = aux.microph_variables
 
    @. source_term = some_micro_foo(stored_variable)
    @. aux.important_sources = to_sources(-source_term, source_term)
end

@inline function important_sources_tendency!

    precompute_aux_important_sources!
    
    @. dY.ρq_tot += aux.thermo_variables.ρ * aux.important_sources.q_tot
end
```

It is more complicated for Cloudy than the other microphysics models. There are software office hours on Wednesday. If we can't figure it out by then, maybe we could submit it as a request to help then?